File: src/main/java/com/study/csdojo/Problem1_MostFrequent.java
package com.study.csdojo;

import java.util.HashMap;
import java.util.Map;

public class Problem1_MostFrequent {

    /**
     * Find the most frequent occuring item in an array.
     */
    Integer mostFrequent(int[] array) {
        int maxCount = -1;
        Integer maxItem = null;
        Map<Integer, Integer> count = new HashMap<>();

        for (int i : array) {
            if (count.containsKey(i)) {
                count.put(i, count.get(i) + 1);
            } else {
                count.put(i, 1);
            }
            if (count.get(i) > maxCount) {
                maxCount = count.get(i);
                maxItem = i;
            }
        }
        return maxItem;
    }

}

File: src/main/java/com/study/csdojo/Problem1_MostFrequent.py
def most_frequent(given_list):
    max_count = -1
    max_item = None
    count = {}

    for i in given_list:
        if i not in count:
            count[i] = 1
        else:
            count[i] += 1
        if (count[i] > max_count):
            max_count = count[i]
            max_item = i
    return max_item

if __name__ == '__main__':
    print("Result of %s must be %i" % (most_frequent([1, 3, 1, 3, 2, 1]), 1))
    print("Result of %s must be %i" % (most_frequent([3, 3, 1, 3, 2, 1]), 3))
    print("Result of %s must be None" % (most_frequent([])))
    print("Result of %s must be %i" % (most_frequent([0]), 0))
    print("Result of %s must be %i" % (most_frequent([0, -1, 10, 10, -1, 10, -1, -1, -1, 1]), -1))